nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -144,7 +144,11 @@ bool Controller::isTrajectoryCollisionFree(
   }
 
   // Generate path
-  for (double t = 0; t < projection_time_; t += simulation_time_step_) {
+  double distance = std::numeric_limits<double>::max();
+  const double resolution = v_linear_max_ * simulation_time_step_;
+  unsigned int max_iter = static_cast<unsigned int>(projection_time_ / simulation_time_step_);
```

Comment:
I'd make sure to `ceil` the output of the division so that if there are fractional timesteps, we use all. Casting rounds down always.